Billing.API/Services/Invoice/InvoiceService.cs

```diff
@@ -22,6 +23,67 @@ public InvoiceService(ILogger<InvoiceService> logger, IOptions<InvoiceProviderOp
         }
 
         public async Task<IEnumerable<InvoiceListItem>> GetInvoices(string clientPrefix, int clientId)
+        {
+            var dt = await GetInvoiceRecords(clientPrefix, clientId);
+
+            return dt.AsEnumerable().Select(dr => new InvoiceListItem(
+                clientPrefix,
+                clientId.ToString(),
+                dr.Field<DateTime>("SendDate").ToDateTimeOffSet(),
+                dr.Field<string>("DocCur"),
+                dr.Field<decimal>("DocTotal").ToDouble(),
+                $"invoice_{dr.Field<DateTime>("SendDate"):yyyy-MM-dd}_{dr.Field<int>("AbsEntry")}.{dr.Field<string>("FileExt")}")).ToList();
+        }
+
+        public async Task<byte[]?> GetInvoiceFile(string clientPrefix, int clientId, int fileId)
+        {
+            var dt = await GetInvoiceRecords(clientPrefix, clientId, fileId);
+
+            var invoice = dt.AsEnumerable().Select(dr => new InvoiceListItem(
+                clientPrefix,
+                clientId.ToString(),
+                dr.Field<DateTime>("SendDate").ToDateTimeOffSet(),
+                dr.Field<string>("DocCur"),
+                dr.Field<decimal>("DocTotal").ToDouble(),
+                $"{dr.Field<string>("trgtPath")}\\{dr.Field<string>("FileName")}.{dr.Field<string>("FileExt")}")).FirstOrDefault();
```

Comment:
Since you only need the _link_, what do you think about avoid creating the InvoiceListItem?

```suggestion
            var invoiceFilename = dt.AsEnumerable().Select(dr => $"{dr.Field<string>("trgtPath")}\\{dr.Field<string>("FileName")}.{dr.Field<string>("FileExt")}").FirstOrDefault();
```

Reply:
Sure